### PR TITLE
python310Packages.b2sdk: 1.19.0 -> 1.24.0; backblaze-b2: 3.7.0 -> 3.9.0

### PR DESCRIPTION
**Pull request by natsukium**
## Description of changes

Changelog: https://github.com/Backblaze/b2-sdk-python/blob/v1.24.0/CHANGELOG.md

Changelog: https://github.com/Backblaze/B2_Command_Line_Tool/blob/v3.9.0/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 253275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages failed to build:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
    <li>python310Packages.b2sdk</li>
    <li>python310Packages.b2sdk.dist</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/b2sdk/default.nix
Comment (kevincox) [inline pkgs/development/python-modules/b2sdk/default.nix:1]: I'm getting a build error on Linux:

[b2.log](https://github.com/NixOS/nixpkgs/files/12515178/b2.log)

**Comment by natsukium** [comment]
<details><summary>Error Log on Linux</summary>

```
=================================== FAILURES ===================================
_________ TestSetLocaleContextManager.test_set_locale_context_manager __________
self = <test.unit.b2http.test_b2http.TestSetLocaleContextManager testMethod=test_set_locale_context_manager>
    def test_set_locale_context_manager(self):
        test_locale = locale.normalize(
            'C.utf8'
        )  # C.UTF-8 on Ubuntu 18.04 Bionic, C.utf8 on Ubuntu 22.04 Jammy
        other_locale = 'C'
    
        saved = locale.setlocale(locale.LC_ALL)
        if saved == test_locale:
            test_locale, other_locale = other_locale, test_locale
    
        locale.setlocale(locale.LC_ALL, other_locale)
>       with setlocale(test_locale):
test/unit/b2http/test_b2http.py:319: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/pkj7cgmz66assy7l18zc7j992npb41nx-python3-3.10.12/lib/python3.10/contextlib.py:135: in __enter__
    return next(self.gen)
b2sdk/b2http.py:69: in setlocale
    yield locale.setlocale(locale.LC_ALL, name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
category = 6, locale = 'en_US.UTF-8'
    def setlocale(category, locale=None):
    
        """ Set the locale for the given category.  The locale can be
            a string, an iterable of two strings (language code and encoding),
            or None.
    
            Iterables are converted to strings using the locale aliasing
            engine.  Locale strings are passed directly to the C lib.
    
            category may be given as one of the LC_* values.
    
        """
        if locale and not isinstance(locale, _builtin_str):
            # convert to string
            locale = normalize(_build_localename(locale))
>       return _setlocale(category, locale)
E       locale.Error: unsupported locale setting
/nix/store/pkj7cgmz66assy7l18zc7j992npb41nx-python3-3.10.12/lib/python3.10/locale.py:620: Error
```
</details>

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 253275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
    <li>python310Packages.b2sdk</li>
    <li>python310Packages.b2sdk.dist</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
Thanks for checking!
I've fixed it.

**Review by tomhoule — APPROVED** [review]
Nice!

(edit: nixpkgs-review succeeded, x86 linux)

**Comment by kevincox** [comment]
Result of `nixpkgs-review pr 253275` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
    <li>python310Packages.b2sdk</li>
    <li>python310Packages.b2sdk.dist</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
  </ul>
</details>

**Pull request merged**